### ISSUE TITLE
Can we support deploying from tags?

### ISSUE THREAD

**Issue by dancrumb** [actions/deploy-pages#76]
I have the following workflow:

```
# Simple workflow for deploying static content to GitHub Pages
name: Deploy static content to Pages

on:
  # Runs on pushes targeting the default branch
  push:
    tags:
      - "v*.*.*"

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
permissions:
  contents: read
  pages: write
  id-token: write

# Allow one concurrent deployment
concurrency:
  group: "pages"
  cancel-in-progress: true

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Use Node.js 18.x
        uses: actions/setup-node@v3
        with:
          node-version: 18.x
          cache: "yarn"
      - name: Setup Pages
        uses: actions/configure-pages@v2
      - name: Setup yarn
        run: yarn
      - name: Build docs
        run: yarn run docs
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v1
        with:
          path: "./docs"

  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    needs:
      - build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v1
```

I created a tag and the job failed with no logs and the following on the UI:

```
Branch "v2.0.2" is not allowed to deploy to github-pages due to environment protection rules.
```

and 

```
The deployment was rejected or didn't satisfy other protection rules.
```

It would be great to get support for deploying docs on a tag

**Comment by yoannchaudet** [comment]
👋 That's more of an Actions problem. What is missing is the concept of protected tags on environments.

![image](https://user-images.githubusercontent.com/14911070/202343832-8dcc3bfb-a38e-4336-91fd-d9b5cb2338c7.png)

We cannot allow Pages deployments on non-protected environments (this is our security paradigm for Actions based deployments).

No ETA yet but we will be adding support for that.

A workaround for now could be to have two workflows:

- A first one that runs on tag creation like you have, run a build and push content to a branch
- A second one that does the deployment from the branch

**Comment by yoannchaudet** [comment]
That's why it's called a workaround 😅

**Comment by till** [comment]
@yoannchaudet is it enough to break this into two workflows?

E.g. I use [upload-pages-artifact](https://github.com/actions/upload-pages-artifact) with a workflow that runs on `on.push.tags` and then start/trigger another workflow once that completes?

I was thinking, or `on.workflow_dispatch` or something like a `on.workflow_run` (with `types: ["completed"]` filter).

**Comment by till** [comment]
@yoannchaudet Amaze! It worked. 🎉 

For whoever else needs it, I ended up with a variation of [this starter workflow](https://github.com/actions/starter-workflows/blob/4a8f18e34dd13d2b6ee4d8da2ba72629eafe1609/pages/jekyll-gh-pages.yml) as I had to trigger jekyll as well now.

TL;DR — do my release in one job and then start another with the correct token permissions to update the page.
